docs/docs/tutorials/auth/add_auth_server.md

```diff
@@ -0,0 +1,337 @@
+            user = response.json()
+            return {
+                "identity": user["id"],  # Unique user identifier
+                "email": user["email"],
+                "is_authenticated": True,
+            }
+    except Exception as e:
+        raise Auth.exceptions.HTTPException(status_code=401, detail=str(e))
+
+
+# Keep our resource authorization from the previous tutorial
+@auth.on
+async def add_owner(ctx, value):
+    """Make resources private to their creator using resource metadata."""
+    filters = {"owner": ctx.user.identity}
+    metadata = value.setdefault("metadata", {})
+    metadata.update(filters)
+    return filters
+```
+
+The most important change is that we're now validating tokens with a real authentication server. Our authentication handler has the private key for our Supabase project, which we can use to validate the user's token and extract their information.
+
+Let's test this with a real user account!
+
+## Testing Authentication Flow
+
+Create a new file `create_users.py`. This will stand-in for a frontend that lets users sign up and log in.
+
+```python
+import argparse
+import asyncio
+import os
+
+import dotenv
+import httpx
+from langgraph_sdk import get_client
+
+dotenv.load_dotenv()
+
+# Get email from command line
+parser = argparse.ArgumentParser()
+parser.add_argument("email", help="Your email address for testing")
+args = parser.parse_args()
+
+base_email = args.email.split("@")
+email1 = f"{base_email[0]}+1@{base_email[1]}"
+email2 = f"{base_email[0]}+2@{base_email[1]}"
+
+SUPABASE_URL = os.environ["SUPABASE_URL"]
+SUPABASE_SERVICE_KEY = os.environ["SUPABASE_SERVICE_KEY"]
+
+
+async def sign_up(email: str, password: str):
+    """Create a new user account."""
+    async with httpx.AsyncClient() as client:
+        response = await client.post(
+            f"{SUPABASE_URL}/auth/v1/signup",
+            json={"email": email, "password": password},
+            headers={"apiKey": SUPABASE_SERVICE_KEY},
+        )
+        assert response.status_code == 200
+        return response.json()
+
+async def main():
+    # Create two test users
+    password = "secure-password"  # CHANGEME
+    print(f"Creating test users: {email1} and {email2}")
+    await sign_up(email1, password)
+    await sign_up(email2, password)
+
+if __name__ == "__main__":
+    asyncio.run(main())
+```
+
+Then run the setup script:
+
+```shell
+python create_users.py CHANGEME@example.com
+```
+
+!!! tip "About test emails"
+    We'll create two test accounts by adding "+1" and "+2" to your email. For example, if you use "myemail@gmail.com", we'll create "myemail+1@gmail.com" and "myemail+2@gmail.com". All emails will be delivered to your original address.
+
+⚠️ Before continuing: Check your email and click both confirmation links. This would normally be handled by your frontend.
+
+Now let's test that users can only see their own data. Create a new file `test_oauth.py`. This will stand-in for your application's frontend.
```

Comment:
minor: I'd personally prefer just client code instead of a file to copy and paste. I'd probably be running in ipython or jupyter